server/ws/websockets.go

```diff
@@ -114,7 +135,7 @@ func (ws *Server) handleWebSocketOnChange(w http.ResponseWriter, r *http.Request
 	}
 
 	wsSession := websocketSession{
-		client:          client,
+		client:          &wsClient{client, &sync.RWMutex{}},
```

Comment:
If we follow the other suggestion, probably we want to wrap the client ASAP, just before the defer that remove the listener and close it later.

Reply:
Done. Made sure we're using WS connection via our wrapper.